io/disk/fiotest.py

```diff
@@ -58,10 +58,18 @@ def setUp(self):
         """
         default_url = "https://brick.kernel.dk/snaps/fio-git-latest.tar.gz"
         url = self.params.get('fio_tool_url', default=default_url)
+        self.disk_type = self.params.get('disk_type', default='')
         device = self.params.get('disk', default=None)
-        self.disk = disk.get_absolute_disk_path(device)
+        if device and not self.disk_type:
+            self.disk = disk.get_absolute_disk_path(device)
+            if self.disk not in disk.get_all_disk_paths():
+                self.cancel("Missing disk %s in OS" % self.disk)
+        elif self.disk_type == 'nvdimm':
+            self.disk = None
+        else:
+            self.cancel("Please Provide valid disk")
+
```

Comment:
This code change looks good to me
but this needs to be moved after line 84,
As when the code hits the else condition -> Teardown 
that errors out with `ERROR: 'FioTest' object has no attribute 'fio_file`
when `device=None`